devito/pointdata.py

```diff
@@ -36,6 +36,7 @@ def __init__(self, *args, **kwargs):
             self.coordinates = DenseData(name='%s_coords' % self.name,
                                          dimensions=[self.indices[1], d],
```

Comment:
Its the spatial dimension of the coordinates, ie. the `Dimension` object that determines whether we deal with 2D or 3D data. Its never iterated over, since the accesses are unrolled in the interpolation expression, but it's needed to allocate the coordinates data correctly.